cpp/src/strings/convert/convert_hex.cu

```diff
@@ -118,6 +118,86 @@ void dispatch_hex_to_integers_fn::operator()<bool>(column_device_view const&,
   CUDF_FAIL("Output for hex_to_integers must not be a boolean type.");
 }
 
+/**
+ * @brief Functor to convert integers to hexadecimal strings
+ *
+ * @tparam IntegerType The specific integer type to convert from.
+ */
+template <typename IntegerType>
+struct integer_to_hex_fn {
+  column_device_view const d_column;
+  offset_type* d_offsets{};
+  char* d_chars{};
+
+  __device__ void byte_to_hex(uint8_t byte, char* hex)
+  {
+    hex[0] = '0';
+    if (byte >= 16) {
+      uint8_t const hibyte = byte / 16;
+
+      hex[0] = hibyte < 10 ? '0' + hibyte : 'A' + (hibyte - 10);
```

Comment:
Here we potentially write to `hex[0]` (global memory?) multiple times. It should be better to write to a local variable then finally copy to the global memory once.